docs/earthdata-varinfo.ipynb

```diff
@@ -198,6 +198,19 @@
    "source": [
     "In the output above, `/Grid/precipiationCal` has three dimensions: `/Grid/time`, `/Grid/lat` and `/Grid/lon`. These dimension variables also refer to their respective [bounds variables](http://cfconventions.org/Data/cf-conventions/cf-conventions-1.10/cf-conventions.html#cell-boundaries). Because `get_required_variables` is recursive, the bounds attributes are also considered required variables for `/Grid/precipitationCal`.\n",
     "\n",
+    "`earthdata-varinfo` checks for references in known CF Convention metadata attributes that are expected to contain references to other variables. These include:\n",
```

Comment:
Is it earthdata-varinfo checking for references or this particular call to `get_required_variables`?

Reply:
This is a good question. Here's the long answer:

The `VarInfo*` (or perhaps more specifically: `Variable*`) classes will try to identify all the references when a file is initially parsed. But all that means is checking the expected metadata attributes and then trying to fully qualify the references it finds based on the location of the variable making that reference and the value of the metadata attribute. (That is - there is no checking that the variable being referred to exists, and there is no recursive tracing at that point)

`VarInfoBase.get_required_variables` is where the dependencies are traced to get a full set.

So maybe what I'm saying is it's a little of column A and a little of column B.

**TL;DR: Maybe a more accurate update would be:**

> When a `VarInfoFromDmr` or `VarInfoFromNetCDF4` object is instantiated, specific CF Convention metadata attributes of each variable are checked for references references to other variables. These references are then fully qualified to absolute paths for later use by methods such as `VarInfoNetCDF4.get_required_variables`. The metadata attributes expected to contain references to other variables, per the CF Conventions include:

Reply:
I think this more accurate update is better.

Reply:
Cool. I have added that here: [b50c3d](https://github.com/nasa/earthdata-varinfo/pull/35/commits/b50c3da3c9d2e4ecb51a76f4003b0100e9cd34fd)